Building docker images using local registry

### Description

There is a pull rate limit recently introduced by dockerhub, therefore any automated jenkins builds start failing with

Step 3/19 : FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
16:49:27  toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
16:49:27  Makefile:53: recipe for target 'build-image' failed

### Importance

Should-have

### Location
Dockerfile and Makefile

### Suggestions for an improvement

Builds should pass

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,7 @@
+ARG DOCKER_REGISTRY
 ARG GO_VERSION=1.13.8
 ARG ALPINE_VERSION=3.11
-FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}golang:${GO_VERSION}-alpine${ALPINE_VERSION} as go-builder
 
 ARG PROJECT_NAME=zookeeper-operator
 ARG REPO_PATH=github.com/pravega/$PROJECT_NAME
@@ -22,7 +23,9 @@ RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o /src/${PROJECT_NAME} \
     /src/cmd/manager
 
 # =============================================================================
-FROM alpine:${ALPINE_VERSION} AS final
+
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}alpine:${ALPINE_VERSION} AS final
+
 
 ARG PROJECT_NAME=zookeeper-operator
 

--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,6 @@ GIT_SHA=$(shell git rev-parse --short HEAD)
 TEST_IMAGE=$(TEST_REPO)-testimages:$(VERSION)
 DOCKER_TEST_PASS=testzkop@123
 DOCKER_TEST_USER=testzkop
-#DOCKER_REGISTRY=devops-repo.isus.emc.com:8116
 .PHONY: all build check clean test
 
 all: check build

--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ GIT_SHA=$(shell git rev-parse --short HEAD)
 TEST_IMAGE=$(TEST_REPO)-testimages:$(VERSION)
 DOCKER_TEST_PASS=testzkop@123
 DOCKER_TEST_USER=testzkop
-
+#DOCKER_REGISTRY=devops-repo.isus.emc.com:8116
 .PHONY: all build check clean test
 
 all: check build
@@ -50,15 +50,15 @@ build-go:
 		-o bin/$(EXPORTER_NAME)-windows-amd64.exe cmd/exporter/main.go
 
 build-image:
-	docker build --build-arg VERSION=$(VERSION) --build-arg GIT_SHA=$(GIT_SHA) -t $(REPO):$(VERSION) .
+	docker build --build-arg VERSION=$(VERSION) --build-arg DOCKER_REGISTRY=$(DOCKER_REGISTRY) --build-arg GIT_SHA=$(GIT_SHA) -t $(REPO):$(VERSION) .
 	docker tag $(REPO):$(VERSION) $(REPO):latest
 
 build-zk-image:
-	docker build --build-arg VERSION=$(VERSION) --build-arg GIT_SHA=$(GIT_SHA) -t $(APP_REPO):$(VERSION) ./docker
+	docker build --build-arg VERSION=$(VERSION)  --build-arg DOCKER_REGISTRY=$(DOCKER_REGISTRY) --build-arg GIT_SHA=$(GIT_SHA) -t $(APP_REPO):$(VERSION) ./docker
 	docker tag $(APP_REPO):$(VERSION) $(APP_REPO):latest
 
 build-zk-image-swarm:
-	docker build --build-arg VERSION=$(VERSION)-swarm --build-arg GIT_SHA=$(GIT_SHA) \
+	docker build --build-arg VERSION=$(VERSION)-swarm  --build-arg DOCKER_REGISTRY=$(DOCKER_REGISTRY) --build-arg GIT_SHA=$(GIT_SHA) \
 		-f ./docker/Dockerfile-swarm -t $(APP_REPO):$(VERSION)-swarm ./docker
 
 test:

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -8,13 +8,14 @@
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
 
-FROM openjdk:11-jdk
+ARG DOCKER_REGISTRY
+FROM  ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}openjdk:11-jdk
 RUN mkdir /zu
 COPY zu /zu
 WORKDIR /zu
 RUN ./gradlew --console=verbose --info shadowJar
 
-FROM zookeeper:3.6.1
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}zookeeper:3.6.1
 COPY bin /usr/local/bin
 RUN chmod +x /usr/local/bin/*
 COPY --from=0 /zu/build/libs/zu.jar /root/

--- a/docker/Dockerfile-swarm
+++ b/docker/Dockerfile-swarm
@@ -7,6 +7,6 @@
 #
 #     http://www.apache.org/licenses/LICENSE-2.0
 #
-
-FROM zookeeper:3.5.5
+ARG DOCKER_REGISTRY
+FROM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}zookeeper:3.6.1
 COPY zoo.cfg.swarm /conf/zoo.cfg

